lib/backend/syncersv1/updateprocessors/hostendpointprocessor.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors
+
+import (
+	"errors"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/backend/watchersyncer"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+// Create a new SyncerUpdateProcessor to sync HostEndpoint data in v1 format for
+// consumption by both Felix and the BGP daemon.
+func NewHostEndpointUpdateProcessor() watchersyncer.SyncerUpdateProcessor {
+	return NewConflictResolvingCacheUpdateProcessor(apiv2.KindHostEndpoint, convertHostEndpointV2ToV1)
+}
+
+// Convert v2 KVPair to the equivalent v1 KVPair.
+func convertHostEndpointV2ToV1(kvp *model.KVPair) (*model.KVPair, error) {
+	// Validate against incorrect key/value kinds.  This indicates a code bug rather
+	// than a user error.
+	v2key, ok := kvp.Key.(model.ResourceKey)
+	if !ok || v2key.Kind != apiv2.KindHostEndpoint {
+		return nil, errors.New("Key is not a valid HostEndpoint resource key")
+	}
+	v2res, ok := kvp.Value.(*apiv2.HostEndpoint)
+	if !ok {
+		return nil, errors.New("Value is not a valid HostEndpoint resource key")
+	}
+
+	v1key := model.HostEndpointKey{
+		Hostname:   v2res.Spec.Node,
+		EndpointID: v2res.GetName(),
+	}
+
+	var v1value *model.HostEndpoint
+	// Deletion operations will have empty values so skip if empty
+	if v2res != nil {
```

Comment:
This check isn't required actually.. you actually dereference v2Res above to get the Node name so it can't be nil

Reply:
When using the conflict resolving cache the conversion method only gets called for adds/modifies.  The deletes are handled using the data in the cache.